karapace/schema_backup.py

```diff
@@ -64,6 +87,81 @@ def __check_partition_count(topic: str, supplier: Callable[[str], AbstractSet[Pa
         )
 
 
+@contextlib.contextmanager
+def _admin(config: Config) -> KafkaAdminClient:
+    """Creates an automatically closing Kafka admin client.
+
+    :param config: for the client.
+    :raises Exception: if client creation fails, concrete exception types are unknown, see Kafka implementation.
+    """
+
+    @retry(
+        before_sleep=__before_sleep("Kafka Admin client creation"),
+        reraise=True,
+        stop=stop_after_delay(timedelta(minutes=1)),
+        wait=wait_fixed(timedelta(seconds=1)),
+    )
+    def __admin() -> KafkaAdminClient:
+        nonlocal config
```

Comment:
What is wrong with a local function? No better way of making something truly private in Python. 😉 Tenacity also offers `Retryable` which results in more code but does away with the decorator.

Reply:
I think this is fine as inner function, the creation and retry is local here. @aiven-anton suggestion of using `minutes=1` states the resolution of the unit well.

Reply:
Yeah nothing wrong per se here (to reviewer: please don't let this hold back approval/merge), just wanted to point out an opportunity for simplification :)

> What is wrong with a local function? No better way of making something truly private in Python.

This sounds a bit like fighting the language rather than being in harmony with it. Nested functions aren't wrong and they do have their place, but in this case we unnecessarily put the work of creating a function in the runtime path when it can be easily broken out to happen on import-time, when we can do that I think we should see that as the default to go for, and don't use privateness as an argument for "inlining".

> Tenacity also offers Retryable which results in more code but does away with the decorator.

The decoration is fine, I'm just suggesting we don't need an intermediary function to decorate, we can decorate `KafkaAdminClient` directly.

Reply:
Gotcha ✅

The note about `nonlocal config` should still apply though, I don't think it's needed?

Reply:
Removed all `nonlocal` usages where no mutation of the captured variables is happening.